UnitsNet/Extensions/QuantityExtensions.cs

```diff
@@ -10,14 +10,31 @@ namespace UnitsNet;
 /// </summary>
 public static class QuantityExtensions
 {
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="quantity"></param>
+    /// <param name="other"></param>
+    /// <param name="tolerance"></param>
+    /// <returns></returns>
+    public static bool Equals(this IQuantity quantity, IQuantity? other, IQuantity tolerance)
```

Comment:
I'll have to double-check which cases are hitting this, but even if it works as expected - I really don't see how anyone could ever use this.. If you don't know what type of quantity you're working with, how can you select the appropriate tolerance?

I was hoping you'd pick option 3, instead you picked option -1 🤣

Reply:
OK you have me convinced.

There are consumers of the untyped `IQuantity`, so it just seemed to me like a natural thing to support equality checks for, since we do offer this interface. Mostly for completeness sake.

But I totally get your point that passing an untyped tolerance seems very fringe. A custom equality comparer to reuse for different quantities is the best I can come up with, but even then I'm sure a generic typed to either linear/affine/logarithmic would be better.

With the fraction PR and accurate equality, I suppose tolerance equality becomes less of a thing. 

I do think there are some improvements to the type parameters here, to have fewer of them, but it still can be reduced yet more. Let me give it another go and update this PR.